cms/static/js/views/utils/create_library_utils.js

```diff
@@ -1,58 +1,17 @@
 /**
  * Provides utilities for validating libraries during creation.
  */
-define(["jquery", "underscore", "gettext", "js/views/utils/view_utils"],
-    function ($, _, gettext, ViewUtils) {
+define(["jquery", "gettext", "js/views/utils/view_utils", "js/views/utils/create_utils_base"],
+    function ($, gettext, ViewUtils, CreateUtilsFactory) {
         "use strict";
         return function (selectors, classes) {
-            var validateTotalKeyLength, setNewLibraryFieldInErr, hasInvalidRequiredFields,
-                createLibrary, validateFilledFields, configureHandlers;
+            var keyLengthViolationMessage = gettext("The combined length of the organization and library code fields cannot be more than <%=limit%> characters.")
+            var keyFieldSelectors = [selectors.org, selectors.number];
+            var nonEmptyCheckFieldSelectors = [selectors.name, selectors.org, selectors.number];
 
-            var validateRequiredField = ViewUtils.validateRequiredField;
-            var validateURLItemEncoding = ViewUtils.validateURLItemEncoding;
+            CreateUtilsFactory.call(this, selectors, classes, keyLengthViolationMessage, keyFieldSelectors, nonEmptyCheckFieldSelectors);
```

Comment:
Yes, I think this is better-- thanks! You can go ahead and merge. :+1: 

Honestly though, I'm still confused about how the helper methods (for instance, validateTotalKeyLength) end up getting returned. create_utils_base returns them in the function it defines, but I don't see how that gets passed through create_library_utils and create_course_utils. Obviously it is working, but there is something about this pattern of "inheritance" that I'm still not understanding. If you have time to explain it to me, I'd appreciate it. :)